core/src/decoder/mod.rs

```diff
@@ -24,6 +24,11 @@ pub const HASHER_STATE_OFFSET: usize = OP_BITS_OFFSET + NUM_OP_BITS;
 /// Number of hasher columns in the decoder trace.
 pub const NUM_HASHER_COLUMNS: usize = 8;
 
+/// Number of relper registers in the decoder trace.
+pub const NUM_USER_OP_HELPERS: usize = 6;
+
+pub const USER_OP_HELPERS_OFFSET: usize = DECODER_TRACE_OFFSET + 10;
```

Comment:
I'd change this to `HASHER_STATE_OFFSET + 2`. Also, let's add a comment saying that the first two registers of the decoder trace are used by the decoder itself.

When we need to use this variable, we can compute the actual offset as: `DECODER_TRACE_OFFSET + USER_OP_HELPERS_OFFSET`.